rotkehlchen/exchanges/binance.py

```diff
@@ -589,6 +605,121 @@ def _query_lending_balances(
+                        end_ts=query_end_ts,
+                        method='lending/union/interestHistory',
+                        # size determines the length of the data returned,
+                        # 100 is the maximum value possible
+                        additional_options={'lendingType': lending_type, 'size': 100},
+                    )
+                except (RemoteError, BinancePermissionError) as e:
+                    self.msg_aggregator.add_error(
+                        f'Failed to query binance lending interest history between '
+                        f'{query_start_ts} and {query_end_ts}. {str(e)}',
+                    )
+                    return True
+
+                for entry in response:
+                    try:
+                        interest_received = deserialize_asset_amount(entry['interest'])  # noqa: E501
+                        if interest_received == ZERO:
+                            continue
+
+                        timestamp = TimestampMS(deserialize_timestamp_from_intms(entry['time']) * 1000)  # noqa: E501
+                        notes = f'Interest paid from {entry["lendingType"]} {entry["productName"]} savings'  # noqa: E501
+                    except KeyError as e:
+                        self.msg_aggregator.add_error(
+                            f'Missing key entry for {str(e)} in {self.name} {entry}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+                    except DeserializationError as e:
+                        self.msg_aggregator.add_error(
+                            f'Error at deserializing {self.name} asset. {str(e)}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+                    try:
+                        asset = asset_from_binance(entry['asset'])
+                    except (UnsupportedAsset, UnknownAsset) as e:
+                        error_type = 'unknown' if isinstance(e, UnknownAsset) else 'unsupported'
+                        self.msg_aggregator.add_warning(
+                            f'Found {error_type} {self.name} asset {e.identifier}. '
+                            f'Ignoring its lending interest history query.',
+                        )
+                        continue
+
+                    event = HistoryBaseEntry(
+                        event_identifier=uuid4().bytes,
+                        sequence_index=0,  # since event_identifier is always different
+                        timestamp=timestamp,
+                        location=self.location,
+                        location_label=self.name,  # the name of the CEX instance
+                        asset=asset,
+                        balance=Balance(
+                            amount=interest_received,
+                            usd_value=ZERO,
```

Comment:
so you put it as zero here and leaveit up to the periodic task to query?

Reply:
Initially, yeah. However, since this request is meant to be async, I'll just make the usd_value request here